src/storage/txn/scheduler.rs

```diff
@@ -309,33 +357,45 @@ impl<E: Engine> Scheduler<E> {
 
     /// Initiates an async operation to get a snapshot from the storage engine, then posts a
     /// `SnapshotFinished` message back to the event loop when it finishes.
-    fn get_snapshot(&mut self, cid: u64) {
-        let task = self.dequeue_task(cid);
+    fn get_snapshot(&self, cid: u64) {
+        let task = self.inner.dequeue_task(cid);
         let tag = task.tag;
         let ctx = task.context().clone();
-        let executor = self.fetch_executor(task.priority());
 
+        let executor = self.fetch_executor(task.priority());
+        let pool = executor.clone_pool();
+        let inner = self.clone();
         let cb = Box::new(move |(cb_ctx, snapshot)| {
             executor.execute(cb_ctx, snapshot, task);
         });
-        if let Err(e) = self.engine.async_snapshot(&ctx, cb) {
-            SCHED_STAGE_COUNTER_VEC
-                .with_label_values(&[tag, "async_snapshot_err"])
-                .inc();
 
-            error!("engine async_snapshot failed"; "err" => ?e);
-            self.finish_with_err(cid, e.into());
-        } else {
-            SCHED_STAGE_COUNTER_VEC
-                .with_label_values(&[tag, "snapshot"])
-                .inc();
-        }
+        pool.schedule(move |sched_ctx: &mut SchedContext<E>| {
```

Comment:
is the flow here:

1. send to the pool through `pool.schedule`
2. in the pool worker, send to local read to get the snapshot through `async_snapshot`
3. resend the snapshot to the pool again through `executor.execute`

if yes, why do we need step 1? seem the previous way didn't use it too.

@hicqu

Reply:
It's for removing local reader. After the local reader thread is removed, snapshots will be get in grpc threads, which is not good. So we can just send the task to scheduler workers.